src/src-assets/creator-assets/controllers.coffee

```diff
@@ -183,29 +193,78 @@ Adventure.controller "AdventureCtrl", ['$scope', '$filter', '$compile', '$rootSc
 	$scope.hideCoverAndModals = ->
 		$scope.showIntroDialog = false
 		$scope.showBackgroundCover = false
+		$scope.showInventoryBackgroundCover = false
 		$scope.nodeTools.show = false
 		$scope.showCreationDialog = false
 		$scope.showDeleteWarning = false
 		$scope.showTitleEditor = false
 		$scope.validation.show = false
 		$scope.showScoreModeDialog = false
 		$scope.showImportTypeSelection = false
+		$scope.showItemSelection = false
+		$scope.showRequiredItems = false
+		$scope.showItemManager = false
+		$scope.showItemIconSelector = false
+		$scope.editingIcons = false
 
 		$scope.resetNewNodeManager()
 
 		$scope.displayNodeCreation = "none"
 
+	$scope.loadIcons = () ->
+		return new Promise((resolve, reject) -> 
+			xhr = new XMLHttpRequest()
+			xhr.open("GET", "assets/icons.json", true)
+			xhr.responseType = 'json'
+			xhr.onload = () ->
+				if xhr.status >= 200 && xhr.status < 300
+					resolve(xhr.response)
+				else
+					reject({
+						status: xhr.status,
+						statusText: xhr.statusText
+					})
+			xhr.onerror = () ->
+				reject({
+					status: xhr.status,
+					statusText: xhr.statusText
+				})
+			xhr.send()
+		)
+
+	$scope.initIcons = (customIcons = null) ->
+		$scope.icons = []
+
+		$scope.loadIcons().then((result) ->
+			angular.forEach result.icons, (icon, index) ->
+				$scope.icons[index] = {
```

Comment:
I'm a little worried about how this works with the scope lifecycle in angular, but assuming it works... it's probably okay. Worst case, we wrap some of this in a `$scope.$apply`, so it doesn't interrupt the scope lifecycle and cause weird race conditions.

Reply:
To reduce loading times, I decided to remove the XMLHttpRequest entirely and manually insert the array of icons (generated by the load_icons.sh script)